### PR TITLE
[wip] Conductor

### PR DESCRIPTION
**Pull request by noorvir**

### CODE REVIEW

File: api/middleware/middleware.go [hunk 1]
```diff
@@ -81,9 +81,9 @@ func GetExecIDFromContext(ctx context.Context) string {
 	return execID
 }
 
-// withAccountCtx is a helper middleware that fakes an authenticated account. It should only
+// WithAccountCtx is a helper middleware that fakes an authenticated account. It should only
 // be user for development or when self-hosting.
```
Comment (cebilon123) [inline api/middleware/middleware.go:85]: *used 😄

File: api/router/exec.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		conductorService: nil,
+	}
+}
+
+func (e *ExecRouter) route(provider types.Provider) *execsrv.Service {
+	switch provider {
+	case types.LambdaLabsProvider:
+		return e.llService
+	case types.UnweaveProvider:
+		panic("not implemented") // this is only available in the hosted version
```
Comment (cebilon123) [inline api/router/exec.go:41]: Wouldn't it be more readable to register it in the map?
Btw. if you think in other way, please explain then we can learn from each other 😄

File: wip/README.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Unweave
+
+The architecture for Unweave is broken into:
+- `Services` at the business logic level,
```
Comment (cebilon123) [inline wip/README.md:4]: What do you think about having an infrastructure layer here as well?

**Review by cebilon123 — COMMENTED** [review]
I know it is WIP but I left some comments.

**Comment by noorvir** [comment]
Closing in favor of a new PR later from the develop branch

**Pull request closed without merging**